dotCMS/src/main/java/com/dotcms/security/secret/SecretsStore.java

```diff
@@ -0,0 +1,70 @@
+package com.dotcms.security.secret;
+
+import java.util.Collection;
+import java.util.Optional;
+import com.dotmarketing.util.Config;
+import io.vavr.control.Try;
+
+public interface SecretsStore {
+
+    /**
+     * Optionally returns the secrets value if there is one, empty if not
+     * 
+     * @param variableKey
+     * @return
+     */
+    public Optional<char[]> getValue(final String variableKey);
+
+    /**
+     * Lists the keys - keys can be case insensitive (forced lowercase) based on the implementation
+     * 
+     * @return
+     */
+    public Collection<String> listKeys();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'public' on method 'listKeys': the method is declared in an interface type](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4623512)